src/stores/widgets/StopGapWidget.ts

```diff
@@ -259,6 +256,15 @@ export class StopGapWidget extends EventEmitter {
             });
         }
     };
+
+    private onRoomViewStoreUpdate = (): void => {
+        const roomId = SdkContextClass.instance.roomViewStore.getRoomId() ?? null;
+        if (roomId !== this.viewedRoomId) {
+            this.messaging!.setViewedRoomId(roomId);
+            this.viewedRoomId = roomId;
+        }
+    };
```

Comment:
Might be worth a comment noting that this listener is only added in certain situations, as this is not the norm.